### ISSUE TITLE
Regression in 17.0.1: uses public registry instead of registry set in .npmrc

### ISSUE THREAD

**Issue by dasa** [raineorshine/npm-check-updates#1439]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)

---

## Steps to Reproduce

.npmrc:

```
@foo:registry = https://x.example.com/artifactory/api/npm/npm-repo

```

Dependencies:

```json

```

Steps:

## Current Behavior

Errors like:

```
@foo/bar 404 Not Found - GET https://registry.npmjs.org/%40foo%2Fbar. All 3 retry attempts failed. Either your internet connection is down, the registry is inaccessible, the authentication credentials are invalid, or the package does not exist. 
```

## Expected Behavior

No errors and uses correct registry.

p.s. - thanks for all the work in this great package!

**Comment by raineorshine** [comment]
Hi. Can you confirm that the error does not exist in `v17.0.0`?

**Comment by destus90** [comment]
In our case this error does not happen in `v17.0.0`

**Comment by raineorshine** [comment]
@swantzter It looks like this was a regression from #1438. Could you look into this?

**Comment by swantzter** [comment]
My guess here is that the registry URL has no trailing slash. new URL() with a base path will place the package relative to last directory, so `new URL(encodeURIComponent('@foo/bar'), 'https://x.example.com/artifactory/api/npm/npm-repo') === 'https://x.example.com/artifctory/api/npm/%40foo%2fbar'`

**Comment by raineorshine** [comment]
@dasa @destus90 @bashbers Let me know if `v17.0.3` fixes it. Thanks!

**Comment by dasa** [comment]
Yes, it's fixed in 17.0.3 😄

**Comment by raineorshine** [comment]
Thanks to @swantzter
